### ISSUE TITLE
Any suggestions???

### ISSUE THREAD

**Issue by AndrewCRMartin** [JamesSweetJones/abYdraw#66]
This is a very strange one that we can't really do with AbML at the moment!

The structure is essentially:
```
VH.a(1:13)-CH1(2:14){1}-H(3:8){2}-CH2(4:9)-CH3(5:10)|VH.a(6:17)-CH1(7:18){1}-H(8:3){2}-CH2(9:4)-CH3(10:5)|X(11)-L(12)-VL.a(13:1)-CL(14:2){1}|X(15)-L(16)-VL.a(17:6)-CL(18:7){1}
```

However: *The X domain is actually inserted within CDR-L1 rather than being at the N-terminus.*

Any suggestions on how we could modify AbML to allow this? I'm thinking something like
```
VH(1:11,15)-CH1(2:16){1}-H(3:8){2}-CH2(4:9)-CH3(5:10)|
VH(6:17,21)-CH1(7:22){1}-H(8:3){2}-CH2(9:4)-CH3(10:5)|
VL'n(11:1,15)-L(12)-X(13)-L(14)-VL'c(15:1,11)-CL(16:2){1}|
VL'n(17:6,21)-L(18)-X(19)-L(20)-VL'c(21:6,17)-CL(22:7){1}
```

**Comment by AndrewCRMartin** [comment]
p.s. I wouldn't worry about trying to display it at the moment, but comments on my suggestion and feasibility ASAP would be good!

**Comment by JamesSweetJones** [comment]
I think that makes logical sense since we've used three kinds of parentheses is AbML already. I'm not too sure about giving a single domain two numbers though. I feel that could get confusing and would appear like there are two domains (although, maybe this would be the case realistically). Here are a couple of suggestions

Option 1 - establish two separate domains (what you suggested)
```
VH(1:11,15)-CH1(2:16){1}-H(3:8){2}-CH2(4:9)-CH3(5:10)|
VH(6:17,21)-CH1(7:22){1}-H(8:3){2}-CH2(9:4)-CH3(10:5)|
VL'n(11:1,15)-L(12)-X(13)-L(14)-VL'c(15:1,11)-CL(16:2){1}|
VL'n(17:6,21)-L(18)-X(19)-L(20)-VL'c(21:6,17)-CL(22:7){1}
```

Option 2 - same domain numbering for c and n regions but out of order numbering

```
VH(1:11)-CH1(2:15){1}-H(3:8){2}-CH2(4:9)-CH3(5:10)|
VH(6:16)-CH1(7:20){1}-H(8:3){2}-CH2(9:4)-CH3(10:5)|
VL'n(11:1)-L(12)-X(13)-L(14)-VL'c(11:1)-CL(15:2){1}|
VL'n(16,6)-L(17)-X(18)-L(19)-VL'c(16:6)-CL(20:7){1}
```

Option 3 - sub domains (a,b,c etc.) within a single domain

```
VH(1:11)-CH1(2:12){1}-H(3:8){2}-CH2(4:9)-CH3(5:10)|
VH(6:13)-CH1(7:14){1}-H(8:3){2}-CH2(9:4)-CH3(10:5)|
VL'n(11:1)-L(11a)-X(11b)-L(11c)-VL'c(11:1)-CL(12:2){1}|
VL'n(13,6)-L(13a)-X(13b)-L(13c)-VL'c(13:6)-CL(14:7){1}
```

Let me know your thoughts. Happy to jump on a call and discuss

**Comment by AndrewCRMartin** [comment]
I'm on my way into London to see Jethro Tull at the Palladium....

Actually I think I prefer your second suggestion. I guess from the abYdraw
point of view that might also be the most logical since the split domain
would still be drawn as a single domain (so just one domain number) with
the insertion being added as a single domain with 2 linkers going to VL.
Does that make sense?

A.

**Comment by JamesSweetJones** [comment]
That makes sense. I think I'm going to do a whole reworking of abYdraw  at this rate. There's been a lot of scope creep as to what was coming through the pipeline then and now.

**Comment by AndrewCRMartin** [comment]
Absolutely, but it's something I was thinking of doing when I get time! I
would do it:
a) using an HTML canvas in JavaScript
b) fully object oriented - not something I usually favour, but here it
really makes sense as each domain is an object with a set of properties and
methods
c) using a minimisation function such that the domains position themselves
automatically. Essentially you want each domain to be as far as possible
from each other domain except when indicated as touching or linked, but you
also want to avoid links/hinges being at more than 45 degrees from a domain.

Taking that sort of approach (with minimal tweaking/optimization), you
should be able to render anything (famous last words!)

A.

**Comment by AndrewCRMartin** [comment]
Redoing it is clearly not something you should consider until you thesis is
submitted!!!!

A.
